schema.json

```diff
@@ -0,0 +1,214 @@
+{
+  "id": "http://api.codeclimate.com/test-coverage-schema#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+
+  "title": "Test Coverage",
+  "description": "Schema that describes the format of the test coverage payload sent to Code Climate",
+
+  "definitions": {
+    "covered_percent": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 100,
+    },
+    "covered_strength": {
+      "type": "integer",
+      "minimum": 0,
+    },
+
+    "line_counts": {
+      "type": "object",
+      "properties": {
+        "covered": {
+          "description": "Number of covered lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+        "missed": {
+          "description": "Number of uncovered lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+        "total": {
+          "description": "Total lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+      },
+      "required": ["covered", "missed", "total"],
+    },
+
+    "git_sha": {
+      "type": "string",
+      "pattern": "^\d{1,}\.\d{1,}\.\d{1,}",
+    },
+
+    "source_file": {
+      "description": "Meta-data and coverage information for source file under test.",
+      "type": "object",
+      "properties": {
+        "blob_id": {
+          "description": "Blob SHA as reported by git",
+          "$ref": "#/definitions/git_sha"
+        },
+        "coverage": {
```

Comment:
Just reading this schema, I wouldn't know that it's a JSON Array of Integer|Null, stringified. Is there a way to do that with `pattern`, or are we left to only `description` for conveying that fact?